### DISCUSSION TITLE
Extend "Block URL" to a new admin screen

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Capabilities#679] [Feature Requests]
The original request from a user was to put his entire site behind a login screen.

He wanted to block taxonomy archives and other URLs that aren't covered by PublishPress Permissions.

Kevin proposed this:

> 
> I think this would be best implemented in Capabilities Pro. We could make Block URLs a separate Capabilities menu item and handle both wp-admin and front end URLs.
> 
> Blocking a front end URL is not fundamentally different from a wp-admin URL. We just need to test it carefully to prevent unintended blockage by operator error or bug. For example, an empty input box should probably not default to blocking the site url.
> 
> This would also require some design to make entry as convenient as possible. Users should not have to manually enter https://domain.com/wp-admin/, etc. There can be a Front End / Admin selector that causes that part of the URL to be auto-entered.

I don't think this is a bad idea, but wonder if it would scale.

For example, how can a user make sure they've covered all possible URLs?

**Comment by agapetry** [comment]
> how can a user make sure they've covered all possible URLs

Similar to the existing Admin restrictions feature, this would be to block specific URLs.  Putting the whole site behind a login screen would be a different issue.

**Comment by stevejburge** [comment]
Yes, I think we may have two separate issues here?

1) Extending "Block URL" could be very useful and open up more possibilities
2) We still need a good solution for the original user question
